examples/ephemeral_gpu_offloading/load_with_dora.py

```diff
@@ -0,0 +1,90 @@
+# Example script demonstrating the time difference loading a model with a DoRA using ephemeral GPU offloading
+# vs doing it purely on the CPU.
+
+import argparse
+import time
+
+from huggingface_hub import snapshot_download
+from transformers import AutoModelForCausalLM
+
+from peft import PeftModel
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Load a model with DoRA using ephemeral GPU offloading")
+    parser.add_argument("--model", type=str, default="NousResearch/Hermes-2-Pro-Mistral-7B", help="Model to load")
+    parser.add_argument(
+        "--dora",
+        type=str,
+        default="anamikac2708/Mistral-7B-DORA-finetuned-investopedia-Lora-Adapters",
+        help="DoRA to use",
+    )
+    parser.add_argument("--ephemeral_gpu_offload", action="store_true", help="Use ephemeral GPU offloading")
+    parser.add_argument(
+        "--merge_model", action="store_true", help="Merge the model with the DoRA model and save to disk"
+    )
+    args = parser.parse_args()
+
+    peft_model_kwargs = {
+        "ephemeral_gpu_offload": args.ephemeral_gpu_offload,
+        "max_memory": {"cpu": "256GiB"},
+        "device_map": {"": "cpu"},
+    }
+
+    # Predownload
+    try:
+        snapshot_download(repo_id=args.model)
+    except Exception as e:
+        print(f"Failed to download model: {e}")
+        # We continue anyway as this might be e.g. a local directory or something
+    try:
+        snapshot_download(repo_id=args.dora)
+    except Exception as e:
+        print(f"Failed to download DoRA: {e}")
+        # We continue anyway as this might be e.g. a local directory or something
+
+    start = time.perf_counter()
+    print("-" * 20 + " Loading model " + "-" * 20)
+    model = AutoModelForCausalLM.from_pretrained(args.model)
+    model_time = time.perf_counter() - start
+    print("-" * 20 + " Loading PeftModel " + "-" * 20)
+    peft_model = PeftModel.from_pretrained(model, args.dora, **peft_model_kwargs)
+    print("-" * 20 + " Done " + "-" * 20)
+    peft_model_time = time.perf_counter() - start
+
+    print(f"Model loading time: {model_time:.2f}s")
+    print(f"PeftModel loading time: {peft_model_time:.2f}s")
+    print(f"Use ephemeral GPU offloading: {args.ephemeral_gpu_offload}")
+
+    if args.merge_model:
+        merged_model = peft_model.merge_and_unload(progressbar=True)
+        merged_model.save_pretrained("merged_model")
+
+
+if __name__ == "__main__":
+    main()
+
+"""
+Example outputs:
```

Comment:
I think it's better to put this docstring to the top of the file and integrate the comment into the docstring. Also, please add the license notice.